### PR TITLE
fix iOS getBatteryLevel value is -1

### PR DESCRIPTION
**Pull request by novain1102**
Re: #https://github.com/cocos/engine-native/issues/4304

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by qiuguohua — APPROVED** [review]

File: native/cocos/platform/ios/modules/Battery.mm [hunk 1]
```diff
@@ -29,6 +29,7 @@ of this software and associated engine source code (the "Software"), a limited,
 namespace cc {
 
 float Battery::getBatteryLevel() const {
+    [UIDevice currentDevice].batteryMonitoringEnabled = YES;
```
Comment (qiuguohua) [inline native/cocos/platform/ios/modules/Battery.mm:33]: Add some comments for better understanding

**Review by qiuguohua — APPROVED** [review]

File: native/cocos/platform/ios/modules/Battery.mm (hunk 1 above)
Comment (novain1102) [reply native/cocos/platform/ios/modules/Battery.mm:33]: done

**Review by qiuguohua — APPROVED** [review]

File: native/cocos/platform/ios/modules/Battery.mm [hunk 2]
```diff
@@ -29,6 +29,8 @@ of this software and associated engine source code (the "Software"), a limited,
 namespace cc {
 
 float Battery::getBatteryLevel() const {
+    // set batteryMonitoringEnabled value to YES otherwise batteryLevel is always -1
+    [UIDevice currentDevice].batteryMonitoringEnabled = YES;
```
Comment (minggo) [inline native/cocos/platform/ios/modules/Battery.mm:33]: Is there any problem that set it more than one time?

Comment (minggo) [reply native/cocos/platform/ios/modules/Battery.mm:33]: Or should check the value first?

Comment (novain1102) [reply native/cocos/platform/ios/modules/Battery.mm:33]: No. we don't need

**Review by minggo — APPROVED** [review]

**Pull request merged**